ProjFS.Mac/PrjFSLib/PrjFSLib.cpp

```diff
@@ -1345,3 +1378,27 @@ static void ReturnFileMutexIterator(FileMutexMap::iterator lockIterator)
         s_fileLocks.erase(lockIterator);
     }
 }
+
+static const char* GetRelativePath(const char* path, const char* root)
```

Comment:
This seems like function prime for unit testing

Reply:
Also is this a copy of the implementation in VirtualizationRoots.cpp?

Reply:
> Also is this a copy of the implementation in VirtualizationRoots.cpp?

Yes, aside from the log statements. With a bit of effort we can actually share that code between kext and userlib, but as mentioned in my introductory comment on the PR under possible further enhancements, I think we should get rid of the kernel side version altogether. We now don't always send a path from kernel to user space, and when we don't, user space generates an absolute path via the system API `fsgetpath()` which it then converts to a relative one. So I'd argue that we're best off sending the absolute path, when available, from the kext too and leave the relative path calculation to the user space lib.

> This seems like function prime for unit testing.

Definitely. So far I've been ~~too lazy for~~ holding off on adding some as it means creating a new target for testing userlib code etc. and I wanted to get the PR into a roughly-final shape first.

Reply:
> So I'd argue that we're best off sending the absolute path, when available, from the kext too and leave the relative path calculation to the user space lib.

I agree it would be nice to consolidate this to user space if we can.  I've filed #846 for this.